AmplifyPlugins/Core/AWSPluginsCore/Model/Decorator/ModelIdDecorator.swift

```diff
@@ -76,17 +76,35 @@ public struct ModelIdDecorator: ModelBasedGraphQLDocumentDecorator {
         if case .mutation = document.operationType {
             var inputMap = [String: String]()
             for (name, value, _) in identifierFields {
-                inputMap[name] = value
+                inputMap[name] = value.graphQLDocumentValue
             }
             inputs["input"] = GraphQLDocumentInput(type: "\(document.name.pascalCased())Input!",
                                                    value: .object(inputMap))
 
         } else if case .query = document.operationType {
             for (name, value, type) in identifierFields {
-                inputs[name] = GraphQLDocumentInput(type: type, value: .scalar(value))
+                inputs[name] = GraphQLDocumentInput(
+                    type: type,
+                    value: identifierFields.count > 1 ? .inline(value) : .scalar(value)
+                )
             }
         }
 
         return document.copy(inputs: inputs)
     }
 }
+
+fileprivate extension ModelIdDecorator {
+    private static func convert(persistable: Persistable) -> GraphQLDocumentValueRepresentable {
+        switch persistable {
+        case let data as Double:
```

Comment:
Double maps to Float scalar I believe but did you need a Float case here?

Reply:
The GraphQL `Float` type is mapped to Swift `Double` type. We don't use Swift `Float` type in codegen.